File: algo/sort/js/insertion.js
const insertion = nums => {
    for (let i = 1; i < nums.length; i++) {

        let j = i;
        while (j > 0 && nums[j] < nums[j - 1]) {
            [ nums[j], nums[j-1] ] = [ nums[j - 1], nums[j] ];
            j--;
        }
    }
}

// test here
let nums = [5, 4, 3, 2, 1, 6, 0];
insertion(nums);
console.log(nums);

File: algo/sort/java/Insertion.java
import java.util.Arrays;

public class Insertion {
    public static void main(String[] args) {
        int[] arr = { 8, 5, 4, 2, 0, -2 };
        Insertion.insertion(arr);
        System.out.println(Arrays.toString(arr));

    }

    public static void insertion(int[] nums) {
        for (int i = 1; i < nums.length; i++) {
            final int val = nums[i];
            int j = i - 1;
            while (j >= 0 && val < nums[j]) {
                nums[j + 1] = nums[j];
                j--;
            }
            nums[j + 1] = val;
        }
    }
}
